### PR TITLE
Use smaller and compressed varients of buttons and form components

### PR DESCRIPTION
**Pull request by AMoo-Miki**
### Description
Replace instances of `EuiButton` that don't have an explicit sizing attribute to `EuiSmallButton*`.
Replace instances of `Eui<form elements>` that don't have density attributes to `EuiCompressed<form elements>`.

### Check List
- [ ] New functionality includes testing.
  - [ ] All tests pass, including unit test, integration test and doctest
- [ ] New functionality has been documented.
  - [ ] New functionality has javadoc added
  - [ ] New functionality has user manual doc added
- [X] Commits are signed per the DCO using --signoff

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by derek-ho** [comment]
@ps48 @mengweieric can one of you folks review this and help get it merged?

File: public/components/event_analytics/explorer/visualizations/config_panel/config_panes/config_controls/config_switch.tsx [hunk 1]
```diff
@@ -14,7 +14,7 @@ interface EUISwitch {
 }
 export const ConfigSwitch: React.FC<EUISwitch> = ({ label, disabled, checked, handleChange }) => (
   <Fragment key={`config-switch-${label}`}>
-    <EuiFormRow label={label}>
+    <EuiCompressedFormRow label={label}>
       <EuiSwitch
```
Comment (derek-ho) [inline public/components/event_analytics/explorer/visualizations/config_panel/config_panes/config_controls/config_switch.tsx:18]: Should this be changed to a compressed switch?

File: public/components/event_analytics/explorer/visualizations/config_panel/config_panes/config_controls/config_treemap_parents.tsx [hunk 2]
```diff
@@ -50,7 +50,7 @@ export const ConfigTreemapParentFields = ({
                     {obj.label !== '' ? obj.label : `Parent ${index + 1}`}
                   </EuiLink>
                 </EuiText>
-                <EuiButtonIcon
+                <EuiSmallButtonIcon
                   color="subdued"
                   iconType="cross"
                   aria-label="clear-field"
```
Comment (derek-ho) [inline public/components/event_analytics/explorer/visualizations/config_panel/config_panes/config_controls/config_treemap_parents.tsx:56]: should icon size prop be removed on line 57?

File: public/components/event_analytics/explorer/visualizations/config_panel/config_panes/config_controls/config_treemap_parents.tsx [hunk 3]
```diff
@@ -64,7 +64,7 @@ export const ConfigTreemapParentFields = ({
       <EuiSpacer size="s" />
       <EuiPanel className="panelItem_button" grow>
         <EuiText size="s">{addButtonText}</EuiText>
-        <EuiButtonIcon
+        <EuiSmallButtonIcon
           iconType="plusInCircle"
           aria-label="clear-field"
           iconSize="s"
```
Comment (derek-ho) [inline public/components/event_analytics/explorer/visualizations/config_panel/config_panes/config_controls/config_treemap_parents.tsx:70]: Same comment as above

File: public/components/visualizations/charts/data_table/data_table_header.tsx [hunk 4]
```diff
@@ -85,15 +85,15 @@ export const GridHeader = ({
         />
       </EuiFlexItem>
       <EuiFlexItem style={{ maxWidth: GRID_HEADER_COLUMN_MAX_WIDTH }}>
-        <EuiButtonEmpty
+        <EuiSmallButtonEmpty
           iconSize="s"
```
Comment (derek-ho) [inline public/components/visualizations/charts/data_table/data_table_header.tsx:89]: remove icon size?

File: public/components/trace_analytics/components/dashboard/latency_trend_cell.tsx [hunk 5]
```diff
@@ -45,7 +45,7 @@ export function LatencyTrendCell({
               <EuiText size="s">{traceGroupName}</EuiText>
             </EuiFlexItem>
             <EuiFlexItem grow={false}>
-              <EuiButtonIcon
```
Comment (derek-ho) [inline public/components/trace_analytics/components/dashboard/latency_trend_cell.tsx:48]: Import should be changed?

**Review by derek-ho — APPROVED** [review]
Minor comments

File: public/components/event_analytics/explorer/visualizations/config_panel/config_panes/config_controls/config_switch.tsx (hunk 1 above)
Comment (AMoo-Miki) [reply public/components/event_analytics/explorer/visualizations/config_panel/config_panes/config_controls/config_switch.tsx:18]: Nah; it already has the compressed attrib.

File: public/components/event_analytics/explorer/visualizations/config_panel/config_panes/config_controls/config_treemap_parents.tsx (hunk 2 above)
Comment (AMoo-Miki) [reply public/components/event_analytics/explorer/visualizations/config_panel/config_panes/config_controls/config_treemap_parents.tsx:56]: No. `EuiSmallButtonIcon` only imposes the `size` prop and `iconSize` should be maintained.

File: public/components/event_analytics/explorer/visualizations/config_panel/config_panes/config_controls/config_treemap_parents.tsx (hunk 3 above)
Comment (AMoo-Miki) [reply public/components/event_analytics/explorer/visualizations/config_panel/config_panes/config_controls/config_treemap_parents.tsx:70]: `EuiSmallButtonIcon` only imposes the `size` prop and `iconSize` should be maintained.

File: public/components/visualizations/charts/data_table/data_table_header.tsx (hunk 4 above)
Comment (AMoo-Miki) [reply public/components/visualizations/charts/data_table/data_table_header.tsx:89]: No.

**Review by mengweieric — APPROVED** [review]

**Comment by ps48** [comment]
@AMoo-Miki Can you please update the 4 snapshots that are failing.

**Comment by TackAdam** [comment]
Can we wait on merging this until https://github.com/opensearch-project/dashboards-observability/pull/2063 gets approved/merged?

**Review by ps48 — APPROVED** [review]

**Pull request merged**